### ISSUE TITLE
[1] rook-ceph - opa policies interfere with rook ceph pods 

### ISSUE THREAD

**Issue by crssnd** [elastisys/compliantkubernetes-apps#620]
**Describe the bug**
rook-ceph-tools and rook-ceph-mgr in the rook-ceph namespace seem to be affected by the OPA policies. maybe other containers also.

**To Reproduce**
Steps to reproduce the behaviour:
1. On a new cluster enable OPA 
2. Clean the docker rook images or restart the pods and check if OPA complains on any breaches 

**Expected behavior**
rook-ceph pods should run with OPA policies enabled 

**Screenshots**
```
2m11s       Warning   FailedCreate           replicaset/rook-ceph-mgr-a-78c5d8cf5d                              Error creating: admission webhook "validation.gatekeeper.sh" denied the request: [denied by require-networkpolicy] No matching networkpolicy found
[denied by require-harbor-repo] container "mgr" has an invalid image repo "ceph/ceph:v15.2.5", allowed repos are ["harbor.test"]
[denied by require-harbor-repo] initContainer "chown-container-data-dir" has an invalid image repo "ceph/ceph:v15.2.5", allowed repos are ["harbor.test"]
[denied by require-harbor-repo] initContainer "init-set-prometheus-server-addr" has an invalid image repo "ceph/ceph:v15.2.5", allowed repos are ["harbor.test"]
13m         Warning   FailedCreate           replicaset/rook-ceph-mgr-a-78c5d8cf5d                              Error creating: admission webhook "validation.gatekeeper.sh" denied the request: [denied by require-harbor-repo] container "mgr" has an invalid image repo "ceph/ceph:v15.2.5", allowed repos are ["harbor.test"]
[denied by require-harbor-repo] initContainer "chown-container-data-dir" has an invalid image repo "ceph/ceph:v15.2.5", allowed repos are ["harbor.test"]
[denied by require-harbor-repo] initContainer "init-set-prometheus-server-addr" has an invalid image repo "ceph/ceph:v15.2.5", allowed repos are ["harbor.test"]
[denied by require-networkpolicy] No matching networkpolicy found
```

**Version (add all relevant versions):**
 - v0.17.0

**Additional context**
should create the policies or add the label to the namespace to exclude it from the checks 

**DoD**

- rook should be able to run with opa enabled

**Comment by viktor-f** [comment]
This can be "fixed" by adding the lable `owner: operator` to the rook namespace, this would make OPA ignore this namespace.
Example for why: https://github.com/elastisys/compliantkubernetes-apps/blob/6d40c7a135a6e2839d498dff9bc009ba61be3ade/helmfile/charts/gatekeeper-constraints/templates/image-registry/constraint.yaml#L18

**Comment by aarnq** [comment]
This also affects user alertmanager and additional services.
Should we add that label as standard for Postgres/Rabbit/Redis as well?

Though I think we need come up with a better solution for user alertmanager, as it is running in a user namespace.

**Comment by OlleLarsson** [comment]
> Though I think we need come up with a better solution for user alertmanager, as it is running in a user namespace.

I am fairly sure that @cristiklein proposed somewhere that it should not be run in a user controller namespace.

**Comment by OlleLarsson** [comment]
> This also affects user alertmanager and additional services. Should we add that label as standard for Postgres/Rabbit/Redis as well?

IMO, yes.

**Comment by cristiklein** [comment]
> > Though I think we need come up with a better solution for user alertmanager, as it is running in a user namespace.
> 
> I am fairly sure that @cristiklein proposed somewhere that it should not be run in a user controller namespace.

:joy: Glad I didn't disappoint you. https://github.com/elastisys/compliantkubernetes-apps/issues/621

TL;DR: The solution I would like us to converge to:

- Run user-alertmanager in the `alertmanager` namespace which is owned by the operator
- Allow the user to change `AlertManagerConfig` in the `alertmanager` namespace via RBAC.

Happy to draft an ADR with namespace labels if that helps. I believe we should not only document "new" decisions, but also "old" decisions.
